public/app/AppWrapper.tsx

```diff
@@ -85,36 +87,48 @@ export class AppWrapper extends React.Component<AppWrapperProps, AppWrapperState
 
     const newNavigationEnabled = Boolean(config.featureToggles.newNavigation);
 
+    const commandPaletteActionSelected = (action: Action) => {
+      reportInteraction('commandPalette_action_selected', {
+        actionId: action.id,
+      });
+    };
+
     return (
       <Provider store={store}>
         <I18nProvider>
           <ErrorBoundaryAlert style="page">
             <ConfigContext.Provider value={config}>
               <ThemeProvider>
-                <ModalsProvider>
-                  <GlobalStyles />
-                  <div className="grafana-app">
-                    <Router history={locationService.getHistory()}>
-                      {newNavigationEnabled ? <NavBarNext /> : <NavBar />}
-                      <main className="main-view">
-                        {pageBanners.map((Banner, index) => (
-                          <Banner key={index.toString()} />
-                        ))}
-
-                        <AngularRoot />
-                        <AppNotificationList />
-                        <SearchWrapper />
-                        {ready && this.renderRoutes()}
-                        {bodyRenderHooks.map((Hook, index) => (
-                          <Hook key={index.toString()} />
-                        ))}
-                      </main>
-                    </Router>
-                  </div>
-                  <LiveConnectionWarning />
-                  <ModalRoot />
-                  <PortalContainer />
-                </ModalsProvider>
+                <KBarProvider
+                  actions={[]}
+                  options={{ enableHistory: true, callbacks: { onSelectAction: commandPaletteActionSelected } }}
+                >
+                  <ModalsProvider>
+                    <GlobalStyles />
+                    {config.featureToggles.commandPalette && contextSrv.isSignedIn && <CommandPalette />}
```

Comment:
i'm sorry for not having thought about this before and having wasted your time: one can configure grafana to be publicly accessible as well (meaning you do not need to login to view, or depending on the settings, even edit and admin an instance): https://grafana.com/docs/grafana/latest/auth/grafana/#anonymous-authentication

just tried this and (as expected) the kbar doesn't show up with `auth.anonymous.enabled = true`.

Reply:
btw, i'm also ok to merge the PR in its current state and come back to this later, it's a quite niche use-case anyway and it won't prevent anyone from using Grafana

Reply:
My thought process is we would only want to show the command palette if there are actions to choose from, and we can dial in when we want to add actions. I changed the hook to only add actions once we are not on the login page.